app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/remote/eventApi/eventApiImpls/EventApiImpl.kt

```diff
@@ -1,51 +1,72 @@
 import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.remote.eventApi.IEventApi
-import com.realityexpander.tasky.agenda_feature.common.util.EventId
 import com.realityexpander.tasky.core.data.remote.TaskyApi
+import com.realityexpander.tasky.core.data.remote.utils.getErrorBodyMessage
+import com.realityexpander.tasky.core.util.UuidStr
+import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.json.Json
 import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okio.*
+import java.io.ByteArrayOutputStream
+import java.util.*
 import javax.inject.Inject
 
+
+@OptIn(ExperimentalSerializationApi::class)
+val jsonPrettyPrint = Json {
+    prettyPrint = true
+    ignoreUnknownKeys = true
+    prettyPrintIndent = "   "
+    encodeDefaults = true
+}
+
+const val MAX_IMAGE_SIZE = 1000000 // 1 MB
+
 class EventApiImpl @Inject constructor(
-    private val taskyApi: TaskyApi
+    private val taskyApi: TaskyApi,
+    private val context: Context
 ) : IEventApi {
+
     override suspend fun createEvent(event: EventDTO.Create): EventDTO.Response {
         try {
             val response = taskyApi.createEvent(
                 createEventRequest =
                     MultipartBody.Part
                         .createFormData(
                             "create_event_request",
-                            Json.encodeToString(EventDTO.Create.serializer(), event),
+                            jsonPrettyPrint.encodeToString(EventDTO.Create.serializer(), event),
                         ),
-                photos = event.photos.map {  // todo - add photo handling
-                    MultipartBody.Part.createFormData("photos", "")
-                },
-//                photos = event.photos.mapIndexed { index, photo -> // todo - add photo handling
-//
-//                val photoFile = Uri.fromFile(  // todo possible solution
-//                    File(
-//                        context.cacheDir,
-//                        context.contentResolver.getFileName(event.photos[0].uri!!)
-//                    )
-//                    ).toFile()
-//
-//                    MultipartBody.Part
-//                        .createFormData(
-//                            "photos",
-//                            "photo$index",
-//                            body = photoFile.toRequestBody() // todo convert URI to ByteArray
-//                        )
-//                }.also {
-//                }
+                photos = event.photos.mapIndexed { index, photo -> // todo - add photo handling
+
+                    val bytes = getBytesFromUri(context, photo.uri)?.also { bytes ->
+                        val size = bytes.size
+                        if (size > MAX_IMAGE_SIZE) {
+                            throw NetworkErrorException("Photo too large")
+                        }
+                    } ?: throw Exception("Photo not found")
```

Comment:
If the user tries to upload 10 photos and just one is too large, this error won't tell them a lot about what they should do (since they don't know which one is too large). I think one of these make sense:
1. Check the photo size when the user tries to add it (maybe you already do that, then ignore this comment). Downside: You'll need to read the bytes of it before you know you'll need them for the upload (however, the size might also be contained in the metadata, not sure right now)
2. You only upload those photos that are actually small enough and let the user know something like `Some photos were too large to be uploaded and were ignored`

Reply:
OK, I will refactor this for better error handling.